update.go

```diff
@@ -381,3 +401,45 @@ other options are ignored.
 		return container.Set(config)
 	},
 }
+
+func upsertWeightDevice(devices []*cgroups.WeightDevice, wd specs.LinuxWeightDevice) []*cgroups.WeightDevice {
+	// Iterate backwards because in case of a duplicate
+	// the last one will be used.
```

Comment:
I've seen the comment, the commit msg, etc. I might be tired, but I don't see why to do it in reverse order or not.

If it's duplicated, the first value must take precedence instead of the last value? Is this on the spec? Intuitively, it seems like an error to me (the config.json contradicts itself).

Reply:
To add a bit of context, this was added as a response to @lifubang suggestion at: https://github.com/opencontainers/runc/pull/4775#discussion_r2141690542

The spec doesn't tell anything about duplicated entries (meaning that they are implicitly allowed, and the precedence rules are not documented either).

The runc behavior is/was to apply all rules one-by-one, so if there is a duplicated rule, the latter one overwrites the former. This is the reason why we iterate backwards.

In practice, though, it doesn't matter much, because if there are no duplicated entries, the order doesn't matter, and my infinite belief in human decency tells me no one has duplicated entries anyway. Yet, if someone does, this trick here saves them from a disaster.

Reply:
You can also run the test case from the second commit without the fix from the same commit to see what's happening if we iterate as usual.